### PR TITLE
feat: Add swapseats command to swap two players' seats with validation

### PR DESCRIPTION
**Pull request by ratteler50**
This pull request introduces a new `swapseats` command to allow storytellers to swap the seating positions of two players in the game. It also includes comprehensive tests to validate the functionality and edge cases of the new command. The most important changes are grouped into functionality additions and testing enhancements.

### Functionality Additions:
* **New `swapseats` Command**: Added the `swapseats` command to swap the seating positions of two players. The command verifies input validity, prevents swapping a player with themselves, updates the seating order, and notifies storytellers of the action. It also creates a backup of the game state after execution. (`commands/player_management_commands.py`, [commands/player_management_commands.pyR223-R264](diffhunk://#diff-56fae1d3b98905b5a2b094b48f0974c7ad7ea022915fe0150f2b2dbb4ef427d5R223-R264))

### Testing Enhancements:
* **Unit Tests for `swapseats` Command**: Added multiple tests to validate the `swapseats` command, covering scenarios such as successful swaps, invalid player arguments, swapping the same player, no active game, and permission denial for non-storytellers. (`tests/commands/test_player_commands.py`, [tests/commands/test_player_commands.pyR1091-R1301](diffhunk://#diff-d1290e2be4fe20c63c61f1fcb7b1ff77623c352da567474de1df0596e3e0c543R1091-R1301))

### CODE REVIEW

File: commands/player_management_commands.py [hunk 1]
```diff
@@ -217,6 +220,48 @@ async def reseat_command(message: discord.Message, argument: str):
...
+    new_seating = game.seatingOrder.copy()
+    index1, index2 = new_seating.index(player1), new_seating.index(player2)
+    new_seating[index1], new_seating[index2] = new_seating[index2], new_seating[index1]
+
+    await game.reseat(new_seating)
+    await message_utils.notify_storytellers_about_action(
+        message.author, f"Swapped seats of {player1.user.name} and {player2.user.name}."
+    )
+
+    if game is not model.game.NULL_GAME:
```
Comment (ratteler50) [inline commands/player_management_commands.py:261]: I'm not sure if this is actually needed since required_phases validates there is an active game

File: commands/player_management_commands.py [hunk 2]
```diff
@@ -217,6 +220,48 @@ async def reseat_command(message: discord.Message, argument: str):
...
+        game_utils.backup("current_game.pckl")
+
+
```
[the remark this answers is not in the document]
Comment (ratteler50) [reply commands/player_management_commands.py:264]: Copilot is just incorrect here :). Goal is to backup the changes, not before making teh changes in case they have issues

File: commands/player_management_commands.py [hunk 3]
```diff
@@ -217,6 +220,48 @@ async def reseat_command(message: discord.Message, argument: str):
...
+        return
+
+    # Swap the players in the seating order
+    new_seating = game.seatingOrder.copy()
+    index1, index2 = new_seating.index(player1), new_seating.index(player2)
+    new_seating[index1], new_seating[index2] = new_seating[index2], new_seating[index1]
+
+    await game.reseat(new_seating)
+    await message_utils.notify_storytellers_about_action(
+        message.author, f"Swapped seats of {player1.user.name} and {player2.user.name}."
```
Comment (jking-roar) [inline commands/player_management_commands.py:258]: lowercase s for action, as this concatenates

Also, should be user.display_name?

**Review by jking-roar — APPROVED** [review]
minor suggestion to update the text of the message sent to all the storytellers
